ServerCore/Pages/Submissions/AuthorIndex.cshtml

```diff
@@ -21,18 +21,26 @@
 <h2>@if (Model.Puzzle != null) { <text>@Model.Puzzle.Name:</text> } Submissions @if (Model.Team != null) { <text>by @Model.Team.Name</text> }</h2>
 <a asp-page="/Puzzles/Index">Back to puzzle list</a> | <a asp-page="/Submissions/SubmissionsWithoutResponses" asp-route-puzzleId="@Model.Puzzle?.ID">Submissions without responses</a>
 <div>
-    @if(Model.Puzzle != null) { <a asp-page="./AuthorIndex" asp-route-puzzleId="" asp-route-teamId="@Model.Team?.ID" asp-route-hideFreeform="@Model.HideFreeform">Clear Puzzle Filter</a> }
+    @if(Model.Puzzle != null) { <a asp-page="./AuthorIndex" asp-route-puzzleId="" asp-route-teamId="@Model.Team?.ID" asp-route-hideFreeform="@Model.HideFreeform" asp-route-favoritesOnly="@Model.FavoritesOnly">Clear Puzzle Filter</a> }
     @if(Model.Puzzle != null && Model.Team != null) { <text>|</text> }  
-    @if(Model.Team != null) { <a asp-page="./AuthorIndex" asp-route-puzzleId="@Model.Puzzle?.ID" asp-route-teamId="" asp-route-hideFreeform="@Model.HideFreeform">Clear Team Filter</a> }  
-    @if(Model.Puzzle != null && Model.Team != null) { <text>|</text> <a asp-page="./AuthorIndex" asp-route-puzzleId="" asp-route-teamId="" asp-route-hideFreeform="@Model.HideFreeform">Clear All Filters</a> }  
+    @if(Model.Team != null) { <a asp-page="./AuthorIndex" asp-route-puzzleId="@Model.Puzzle?.ID" asp-route-teamId="" asp-route-hideFreeform="@Model.HideFreeform" asp-route-favoritesOnly="@Model.FavoritesOnly">Clear Team Filter</a> }  
+    @if(Model.Puzzle != null && Model.Team != null) { <text>|</text> <a asp-page="./AuthorIndex" asp-route-puzzleId="" asp-route-teamId="" asp-route-hideFreeform="@Model.HideFreeform" asp-route-favoritesOnly="false">Clear All Filters</a> }  
     <text>|</text>
     @if (Model.HideFreeform)
     {
-        <a asp-page="./AuthorIndex" asp-route-puzzleId="@Model.Puzzle?.ID" asp-route-teamId="@Model.Team?.ID" asp-route-hideFreeform="false" asp-route-sort="@Model.Sort">Show Freeform</a>
+        <a asp-page="./AuthorIndex" asp-route-puzzleId="@Model.Puzzle?.ID" asp-route-teamId="@Model.Team?.ID" asp-route-hideFreeform="false" asp-route-sort="@Model.Sort">Show Freeform</a> 
     }
     else
     {        
-        <a asp-page="./AuthorIndex" asp-route-puzzleId="@Model.Puzzle?.ID" asp-route-teamId="@Model.Team?.ID" asp-route-hideFreeform="true" asp-route-sort="@Model.Sort">Hide Freeform</a>
+        <a asp-page="./AuthorIndex" asp-route-puzzleId="@Model.Puzzle?.ID" asp-route-teamId="@Model.Team?.ID" asp-route-hideFreeform="true" asp-route-sort="@Model.Sort">Hide Freeform</a> 
+    }
+    @if (!Model.FavoritesOnly || Model.HideFreeform)
+    {
+        <a asp-page="./AuthorIndex" asp-route-puzzleId="@Model.Puzzle?.ID" asp-route-teamId="@Model.Team?.ID" asp-route-hideFreeform="false" asp-route-favoritesOnly="true" asp-route-sort="@Model.Sort">| Freeform Favorites</a>
```

Comment:
If we're adding favorites, might as well allow favoriting of wrong answers in non-freeform cases?

Reply:
Though any tweaks here can wait if it's the only thing to address (and if it is indeed advisable at all). I'm more concerned about the shape of the data and the risk level in touching the DB at all, just because of the timeframe.

Reply:
BTW, "favoriting of wrong answers" is exactly the workflow I want (as an author) in place of the "secret achievements" discussion we are having. I mark as many as I like, I am not in competition with other puzzle authors to use them, feedback can be immediate, etc.

Reply:
(Not that I think we have runway to do it today, mind you)

Reply:
Agreed it sounds good. Made the freeform and favorite filtering orthoganal on this page, but didn't try to add a way to favorite wrong answers

Reply:
Not renaming the DB column though to avoid any risk of messing up the migration

Reply:
Also, to fully support favoriting of wrong answers/freeform, the player index should show what has been favorited.